src/osemgrep/cli_ui/Console_Spinner.ml

```diff
@@ -4,18 +4,25 @@
 
 let spinner = [| "⠋"; "⠙"; "⠹"; "⠸"; "⠼"; "⠴"; "⠦"; "⠧"; "⠇"; "⠏" |]
 
+(* only show the console spinner if this is a non-JSOO Unix TTY *)
+let should_show_spinner () =
+  !ANSITerminal.isatty Unix.stdout && (not !Common.jsoo) && Sys.unix
+
 let show_spinner delay_ms : unit =
-  let print_frame ~frame_index:i : unit =
-    let spinner = spinner.(i mod Array.length spinner) in
-    ANSITerminal.set_cursor 1 (-1);
-    ANSITerminal.printf [ ANSITerminal.green ] "%s Waiting for sign in..."
```

Comment:
I think it's not just used for sign-in now, we also use it when we download rules I think, so we probably should make the alt: message a parameter

Reply:
One note is that the sign-in spinner is intended to block program execution, whereas the spinner for indicating that we're loading rules from the registry is intended to be async and promise-based.